CMakeLists.txt

```diff
@@ -22,155 +22,143 @@ cmake_policy(SET CMP0091 NEW)  # MSVC runtime support
 
 project(astcencoder VERSION 3.0.0)
 
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+set(PACKAGE_ROOT astcenc)
+
+include(CTest)
+
 option(ISA_AVX2 "Enable builds for AVX2 SIMD")
 option(ISA_SSE41 "Enable builds for SSE4.1 SIMD")
 option(ISA_SSE2 "Enable builds for SSE2 SIMD")
 option(ISA_NEON "Enable builds for NEON SIMD")
 option(ISA_NONE "Enable builds for no SIMD")
-option(ISA_NATIVE "Enable builds for the native local CPU")
-# Manually provide this as a tristate so we can distinguish the default
-# not set state from explicitly ON and OFF ...
-# set(ISA_NATIVE AUTO CACHE STRING "Enable builds for the native local CPU")
-# set_property(CACHE ISA_NATIVE PROPERTY STRINGS AUTO ON OFF)
+option(DECOMPRESSOR "Enable builds for decompression only")
+option(DIAGNOSTICS "Enable builds for diagnostic trace")
+option(UNITTEST "Enable builds for unit tests")
 
 set(UNIVERSAL_BUILD OFF)
-set(X64_ISA_COUNT 0)
+set(MACOS_BUILD OFF)
+set(MACOS_ARCH_LEN 0)
 
-set(CONFIGS ${ISA_NONE} ${ISA_NATIVE} ${ISA_AVX2} ${ISA_SSE41} ${ISA_SSE2})
+# Preflight for some macOS-specific build options
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+    set(MACOS_BUILD ON)
+    list(LENGTH CMAKE_OSX_ARCHITECTURES MACOS_ARCH_LEN)
```

Comment:
You can't get this untill xcode starts the actual build, way later after the configure step unfortunately.